### PR TITLE
Added support for command-line args for kokkos-tools

### LINKED ISSUE #2966: Create kokkosp_parse_args and kokkosp_print_usage callbacks for profiling interface and kokkos-profiler command line option
So both @DavidPoliakoff and I are not fans of environment variables. They tend to be poorly documented and commonly cause issues when set and forgotten about. Here is the idea:

## Kokkos Command Line Option

Option specified in the form of: `--kokkos-profiler="<ARGS>"` taking one argument since allowing more than one is problematic to parsing. Then Kokkos takes the argument, space delimits into an array, and invokes `kokkosp_parse_args(...)` if provided.

## kokkosp_parse_args

The callback for parsing arguments. Couple options w.r.t. signature but ultimately, I think what should be passed to the function should be the `--kokkos-profiler` argument space delimited into an array and (personally) I would like to be able to get the executable name. Ideally, I think the best option would be to invoke `kokkosp_parse_args` such that from the tool perspective, it was as if it called directly in the code, e.g.

```console
./myexe --kokkosp-profiler="-c wall_clock --papi-events PAPI_LST_INST"
```

would look like this inside of `kokkosp_parse_args`:

```console
./myexe -c wall_clock --papi-events PAPI_LST_INST`
```

with, rather obviously, copies of the command line arguments so any modifications don't actually propagate to the app.

## kokkosp_print_usage

While it may seem this is unnecessary since `kokkosp_parse_args` could detect a `-h/--help` and print the usage, this actually serves a second purpose: if the tool __*requires*__ environment variables before the args are parsed or command line options are not supported, this could serve as a simple documentation of what the tool uses for configuration. Kokkos would simply look for a `-h` or `--help` in `--kokkos-profiler="<ARGS>"` and call it when dlopened regardless of whether `kokkosp_parse_args` is provided.

## Open questions

- Should `kokkosp_parse_args` be invoked before or after `kokkosp_init_library`?
  - Case for before: that is how it is traditionally done -- a tool parses command line arguments and then initializes
  - Case for after: most tools expect the init to be the first thing called so make it second might require migration and backwards compatibility issues requiring logic to determine if the kokkos version being used calls `kokkosp_parse_args`

### PR DESCRIPTION
**Pull request by jrmadsen**
- "--kokkos-tools-library"
  - specify tools library via command-line instead of `KOKKOS_PROFILE_LIBRARY` env variable
- "--kokkos-tools-args"
  - pass arguments to the tool that is loaded
  - the argument should be a single string and Kokkos will prefix it with the executable and delimit the string (so it looks exactly like `(int argc, char** argv)`
- "--kokkos-tools-help"
  - requests that the tool print information about what it will accept as arguments
- Closes #2966 
- Also, minor change in checking for `"KOKKOS_DISABLE_WARNINGS"` environment variable
  - Original case-sensitive match: `if ((env_str == "TRUE") || (env_str == "ON") || (env_str == "1"))`
  - New case-insensitive regex: `regex_match(env_str, regex("^(true|on|yes|[1-9])$", ...))`
- Adds `"COMMON_ARGS"` to `KOKKOS_ADD_TEST` function in CMake
  - the existing `"ARGS"` argument to creates a new test for each argument, when `"COMMON_ARGS"` are specified, these args are appended to every test, e.g.
  - `ARGS --option=A --option=B` would create two tests, one with `--option=A` and one with `--option=B`
  - `ARGS --option=A --option=B COMMON_ARGS --option=C` would create two tests, one with `--option=A --option=C` and one with `--option=B --option=C`

## Kokkos-Tools functions

```cpp
extern "C" void kokkosp_print_help(char* exe);
extern "C" void kokkosp_parse_args(int argc, char** argv);
```

## Reference

```console
$ ./MC_DartSampler 10
#---------------------------------------------------------------------------#
# KokkosP: timemory Connector (sequence is 0, version: 20200625)
#---------------------------------------------------------------------------#

Reference Value for pi:  3.141593
Serial singleton address: 0x10f2ef540
Serial singleton address: 0x10f2ef540
darts = 1024  hits = 798  pi est = 3.117188

#---------------------------------------------------------------------------#
KokkosP: Finalization of timemory Connector. Complete.
#---------------------------------------------------------------------------#

[kokkos_memory]|0> Outputting 'timemory-MC-DartSampler-output/2020-11-28_02.18_AM/kokkos_memory.json'...
[kokkos_memory]|0> Outputting 'timemory-MC-DartSampler-output/2020-11-28_02.18_AM/kokkos_memory.tree.json'...
[kokkos_memory]|0> Outputting 'timemory-MC-DartSampler-output/2020-11-28_02.18_AM/kokkos_memory.txt'...

[wall]|0> Outputting 'timemory-MC-DartSampler-output/2020-11-28_02.18_AM/wall.flamegraph.json'...
[wall]|0> Outputting 'timemory-MC-DartSampler-output/2020-11-28_02.18_AM/wall.json'...
[wall]|0> Outputting 'timemory-MC-DartSampler-output/2020-11-28_02.18_AM/wall.tree.json'...
[wall]|0> Outputting 'timemory-MC-DartSampler-output/2020-11-28_02.18_AM/wall.txt'...

Serial singleton address: 0x10f2ef540
```
## Help Example

```console
$ ./MC_DartSampler --kokkos-tools-help 10
#---------------------------------------------------------------------------#
# KokkosP: timemory Connector (sequence is 0, version: 20200625)
#---------------------------------------------------------------------------#

Usage:  [tim::argparse::argument_parser arguments...] 
    Pass command-line argument to this library via '--kokkos-tools', e.g.:

        ./myexe --kokkos-threads=4 --kokkos-tools="-c wall_clock cpu_clock --flat-profile"

    where "-c wall_clock cpu_clock --flat-profile" are the arguments intended for timemory
    

Options:
    -C, --config                   Configuration file for timemory
    --suppress-config, --no-config Disable processing of setting configuration files
    --suppress-parsing             Disable parsing environment   
    --enabled                      Activation state of timemory  
    --verbose                      Verbosity level               
    --debug                        Enable debug output           
    --flat-profile                 Set the label hierarchy mode to default to flat
    --timeline-profile             Set the label hierarchy mode to default to timeline
    --max-depth                    Set the maximum depth of label hierarchy reporting
    -c, --kokkos-components        A specification of components which will be used by the interfaces which are designed for kokkos profiling.

    ... etc. ...

    -h, -?, --help                 Shows this page               
```

## Usage Example

Note the extra output files (`**/cpu.*`)

```console
$ ./MC_DartSampler --kokkos-tools-args="-c wall_clock cpu_clock" 10
#---------------------------------------------------------------------------#
# KokkosP: timemory Connector (sequence is 0, version: 20200625)
#---------------------------------------------------------------------------#

Reference Value for pi:  3.141593
Serial singleton address: 0x10cb46540
Serial singleton address: 0x10cb46540
darts = 1024  hits = 798  pi est = 3.117188

#---------------------------------------------------------------------------#
KokkosP: Finalization of timemory Connector. Complete.
#---------------------------------------------------------------------------#

[kokkos_memory]|0> Outputting 'timemory-MC-DartSampler-output/2020-11-28_02.18_AM/kokkos_memory.json'...
[kokkos_memory]|0> Outputting 'timemory-MC-DartSampler-output/2020-11-28_02.18_AM/kokkos_memory.tree.json'...
[kokkos_memory]|0> Outputting 'timemory-MC-DartSampler-output/2020-11-28_02.18_AM/kokkos_memory.txt'...

[cpu]|0> Outputting 'timemory-MC-DartSampler-output/2020-11-28_02.18_AM/cpu.flamegraph.json'...
[cpu]|0> Outputting 'timemory-MC-DartSampler-output/2020-11-28_02.18_AM/cpu.json'...
[cpu]|0> Outputting 'timemory-MC-DartSampler-output/2020-11-28_02.18_AM/cpu.tree.json'...
[cpu]|0> Outputting 'timemory-MC-DartSampler-output/2020-11-28_02.18_AM/cpu.txt'...

[wall]|0> Outputting 'timemory-MC-DartSampler-output/2020-11-28_02.18_AM/wall.flamegraph.json'...
[wall]|0> Outputting 'timemory-MC-DartSampler-output/2020-11-28_02.18_AM/wall.json'...
[wall]|0> Outputting 'timemory-MC-DartSampler-output/2020-11-28_02.18_AM/wall.tree.json'...
[wall]|0> Outputting 'timemory-MC-DartSampler-output/2020-11-28_02.18_AM/wall.txt'...

Serial singleton address: 0x10cb46540
```

### CODE REVIEW

File: core/src/impl/Kokkos_Core.cpp [hunk 1]
```diff
@@ -526,7 +565,7 @@ void parse_command_line_arguments(int& narg, char* arg[],
       --kokkos-disable-warnings      : disable kokkos warning messages
       --kokkos-tune-internals        : allow Kokkos to autotune policies and declare
                                        tuning features through the tuning system. If
-				         left off, Kokkos uses heuristics
+				       left off, Kokkos uses heuristics
```
Comment (dalg24) [inline core/src/impl/Kokkos_Core.cpp:568]: Undo this change

File: core/src/impl/Kokkos_Core.cpp [hunk 2]
```diff
@@ -711,7 +756,9 @@ void parse_environment_variables(InitArguments& arguments) {
     for (char& c : env_str) {
       c = toupper(c);
     }
-    if ((env_str == "TRUE") || (env_str == "ON") || (env_str == "1"))
+    const auto _rc = std::regex_constants::icase | std::regex_constants::egrep;
+    const auto _re = std::regex("^(true|on|yes|[1-9])$", _rc);
+    if (std::regex_match(env_str, _re))
```
Comment (dalg24) [inline core/src/impl/Kokkos_Core.cpp:761]: I don't necessarily have an issue with it but this is a change in behavior so you must add it to the description of the PR.

**Review by dalg24 — DISMISSED** [review]
Add unit tests for passing arguments to the tools.

File: core/src/impl/Kokkos_Core.cpp (hunk 2 above)
Comment (jrmadsen) [reply core/src/impl/Kokkos_Core.cpp:761]: Ah right

File: core/src/impl/Kokkos_Serial.cpp [hunk 3]
```diff
@@ -147,7 +149,11 @@ HostThreadTeamData* serial_get_thread_team_data() {
 
 namespace Kokkos {
 
-bool Serial::impl_is_initialized() { return Impl::g_serial_is_initialized; }
+bool Serial::impl_is_initialized() {
+  std::cout << "Serial singleton address: " << &Impl::g_serial_thread_team_data
+            << std::endl;
```
[the remark this answers is not in the document]
Comment (jrmadsen) [reply core/src/impl/Kokkos_Serial.cpp:154]: Oh weird, must have been debugging stuff from something else.

File: core/src/impl/Kokkos_Core.cpp (hunk 1 above)
Comment (jrmadsen) [reply core/src/impl/Kokkos_Core.cpp:566]: Oh thats strange... There must be a `\t` in there because this change aligned it properly.

Comment (jrmadsen) [reply core/src/impl/Kokkos_Core.cpp:566]: Screen Shot 2020-11-28 at 8 35 52 PM

Yep.

**Comment by jrmadsen** [comment]
So it looks like all this passed except the macOS builds never even finished installing their dependencies...

File: core/unit_test/tools/printing-tool.cpp [hunk 4]
```diff
@@ -23,6 +32,16 @@ extern "C" void kokkosp_finalize_library() {
...
 
+extern "C" void kokkosp_print_help(char* exe) {
+  std::cout << "kokkosp_print_help:" << get_basename(exe) << "::";
+}
+
+extern "C" void kokkosp_parse_args(int argc, char** argv) {
+  std::cout << "kokkosp_parse_args:" << argc;
+  for (int i = 0; i < argc; ++i) std::cout << ":" << get_basename(argv[i], i);
+  std::cout << "::";
+}
```
Comment (dalg24) [inline core/unit_test/tools/printing-tool.cpp:43]: You should actually pass arguments to the executable.

Comment (DavidPoliakoff) [reply core/unit_test/tools/printing-tool.cpp:43]: Yeah, so far that's my only problem. We should do this legitimately, run the test with args

**Review by DavidPoliakoff — CHANGES_REQUESTED** [review]
Damien's comment is the only thing preventing me from approving, the rest is solid. Thanks for the contribution

**Review by DavidPoliakoff — APPROVED** [review]
This addressed my comments, and the added functionality (--kokkos-tools-library) looks well implemented. I'm approving, but we shouldn't merge until we hear from @jjwilke or whoever he wants to review build system issues about whether the build changes are okay

File: cmake/fake_tribits.cmake [hunk 5]
```diff
@@ -121,7 +121,7 @@ FUNCTION(KOKKOS_ADD_TEST)
     CMAKE_PARSE_ARGUMENTS(TEST
       "WILL_FAIL"
       "FAIL_REGULAR_EXPRESSION;PASS_REGULAR_EXPRESSION;EXE;NAME;TOOL"
-      "CATEGORIES;ARGS"
+      "CATEGORIES;ARGS;COMMON_ARGS"
```
Comment (dalg24) [inline cmake/fake_tribits.cmake:124]: Why did you introduce `COMMON_ARGS`?  Does `ARGS` not do what you want?

Comment (dalg24) [reply cmake/fake_tribits.cmake:124]: It was pointed it out to me that this is covered in the description of the PR

**Comment by masterleinad** [comment]
```
test 15
      Start 15: KokkosCore_ProfilingTestLibraryLoadHelp
15: Test command: /var/jenkins/workspace/Kokkos/build-tests/core/unit_test/KokkosCore_ProfilingAllCalls "--kokkos-tools-help"
15: Environment variables: 
15:  KOKKOS_PROFILE_LIBRARY=/var/jenkins/workspace/Kokkos/build-tests/core/unit_test/libkokkosprinter-tool.so
15: Test timeout computed to be: 1500
15: Kokkos::Cuda::initialize WARNING: Cuda is allocating into UVMSpace by default
15:                                   without setting CUDA_LAUNCH_BLOCKING=1.
15:                                   The code must call Cuda().fence() after each kernel
15:                                   or will likely crash when accessing data on the host.
15: Kokkos::Cuda::initialize WARNING: Cuda is allocating into UVMSpace by default
15:                                   without setting CUDA_MANAGED_FORCE_DEVICE_ALLOC=1 or 
15:                                   setting CUDA_VISIBLE_DEVICES.
15:                                   This could on multi GPU systems lead to severe performance
15:                                   penalties.
15: kokkosp_init_library::kokkosp_print_help:KokkosCore_ProfilingAllCalls::Kokkos::Cuda ERROR: Failed to call Kokkos::Cuda::finalize()
15/49 Test #15: KokkosCore_ProfilingTestLibraryLoadHelp ......***Failed  Required regular expression not found. Regex=[kokkosp_init_library::kokkosp_print_help:KokkosCore_ProfilingAllCalls::[
]
]  0.35 sec

test 16
      Start 16: KokkosCore_ProfilingTestLibraryCmdLineHelp
16: Test command: /var/jenkins/workspace/Kokkos/build-tests/core/unit_test/KokkosCore_ProfilingAllCalls "--kokkos-tools-help" "--kokkos-tools-library=/var/jenkins/workspace/Kokkos/build-tests/core/unit_test/libkokkosprinter-tool.so"
16: Test timeout computed to be: 1500
16: Kokkos::Cuda::initialize WARNING: Cuda is allocating into UVMSpace by default
16:                                   without setting CUDA_LAUNCH_BLOCKING=1.
16:                                   The code must call Cuda().fence() after each kernel
16:                                   or will likely crash when accessing data on the host.
16: Kokkos::Cuda::initialize WARNING: Cuda is allocating into UVMSpace by default
16:                                   without setting CUDA_MANAGED_FORCE_DEVICE_ALLOC=1 or 
16:                                   setting CUDA_VISIBLE_DEVICES.
16:                                   This could on multi GPU systems lead to severe performance
16:                                   penalties.
16: kokkosp_init_library::kokkosp_print_help:KokkosCore_ProfilingAllCalls::Kokkos::Cuda ERROR: Failed to call Kokkos::Cuda::finalize()
16/49 Test #16: KokkosCore_ProfilingTestLibraryCmdLineHelp ...***Failed  Required regular expression not found. Regex=[kokkosp_init_library::kokkosp_print_help:KokkosCore_ProfilingAllCalls::[
]
]  0.35 sec
```

File: core/src/impl/Kokkos_Core.cpp [hunk 6]
```diff
@@ -270,8 +271,15 @@ void initialize_backends(const InitArguments& args) {
...
 
-void initialize_profiling(const InitArguments&) {
-  Kokkos::Profiling::initialize();
+void initialize_profiling(const InitArguments& args) {
+  Kokkos::Profiling::initialize(args.tool_lib);
+  if (args.tool_help) {
+    if (!Kokkos::Profiling::printHelp(args.tool_args)) {
+      std::cerr << "Tool has not provided a help message" << std::endl;
+    }
+    std::exit(EXIT_FAILURE);
```
Comment (dalg24) [inline core/src/impl/Kokkos_Core.cpp:280]: Why do you print a non-zero error code?  If we end up here it is not an error while parsing arguments.

For reference, neither `--help` nor `--kokkos-help` cause program termination.
I think it is reasonable to terminate here but I will observe it is inconsistent with `--kokkos-help`.  I would be open to change the behavior when `--kokkos-help` is passed (not in this PR).  (I would strongly object to `--help` calling `std::exit` because it plays poorly when composing in user code)

File: cmake/fake_tribits.cmake (hunk 5 above)
Comment (jjwilke) [inline cmake/fake_tribits.cmake:124]: It doesn't look like ARGS is being used with COMMON_ARGS anywhere in the tests? Am I missing a usage somewhere. My instinct would be to not have this argument since there are other ways of adding a common argument to a bunch of tests, but I don't have a strong opinion.

File: core/src/impl/Kokkos_Core.cpp [hunk 7]
```diff
@@ -540,6 +599,18 @@ void parse_command_line_arguments(int& narg, char* arg[],
...
+                                       name of library if the path is present in the
+                                       runtime library search path (e.g. LD_LIBRARY_PATH)
+      --kokkos-tools-help            : Query the (loaded) kokkos-tool for its command-line
+                                       option support (which should then be passed via
+                                       --kokkos-tools-args="...")
+      --kokkos-tools-args=STR        : A single (quoted) string of options which will be
+                                       whitespace delimited and passed to the loaded
+                                       kokkos-tool as command-line arguments. E.g.
+                                       `<EXE> --kokkos-tools-args="-c input.txt"` will
+                                       pass `<EXE> -c input.txt` as argc/argv to tool
```
[the remark this answers is not in the document]
Comment (jrmadsen) [reply core/src/impl/Kokkos_Core.cpp:613]: I wanted to make it such that, from a tool perspective, this was pretty much identical to being called from `main(int argc, char** argv)`. Makes it easy to just use any existing argparser interface a tool might already have.

File: core/src/impl/Kokkos_Profiling.cpp [hunk 8]
```diff
@@ -332,7 +391,9 @@ void initialize() {
 #ifdef KOKKOS_ENABLE_LIBDL
   void* firstProfileLibrary = nullptr;
 
-  char* envProfileLibrary = getenv("KOKKOS_PROFILE_LIBRARY");
+  char* envProfileLibrary = (profileLibrary.empty())
+                                ? getenv("KOKKOS_PROFILE_LIBRARY")
+                                : const_cast<char*>(profileLibrary.c_str());
```
[the remark this answers is not in the document]
Comment (jrmadsen) [reply core/src/impl/Kokkos_Profiling.cpp:396]: I guess, I personally just let explicit things like command-line args override environment variables

File: core/src/impl/Kokkos_Core.cpp [hunk 9]
```diff
@@ -509,6 +538,36 @@ void parse_command_line_arguments(int& narg, char* arg[],
...
+      // instead of:
+      //    "<EXE>" "-c" "my" "example"
+      if (!tool_args.empty()) {
+        if (tool_args.front() == '"') tool_args = tool_args.substr(1);
+        if (tool_args.back() == '"')
+          tool_args = tool_args.substr(0, tool_args.length() - 1);
+      }
+      // add the name of the executable to the beginning
+      if (narg > 0) tool_args = std::string(arg[0]) + " " + tool_args;
+    } else if (check_arg(arg[iarg], "--kokkos-tools-help")) {
```
[the remark this answers is not in the document]
Comment (jrmadsen) [reply core/src/impl/Kokkos_Core.cpp:566]: Nah I don't think so. I have a lot of options in timemory and I know other tools like Score-P and TAU do too. I think automatically printing that might flood the user with information they might not want

File: core/src/impl/Kokkos_Core.cpp (hunk 6 above)
Comment (jrmadsen) [reply core/src/impl/Kokkos_Core.cpp:280]: > Why do you print a non-zero error code?

Not sure what you mean by "print" here. I feel like not exiting when any flavor of "--help" is requested from the command-line is quite non-standard. I honestly can't think of a single time when I would pass "--help" to a program and have it continue running. 

> I would strongly object to --help calling std::exit because it plays poorly when composing in user code

Yeah, I have generally adopted the LLDB methodology for grouping arguments, if there is a `--` somewhere, I assume the arguments up the first `--` are intended for me and everything after is intended for the app, e.g. `timem ls -t` doesn't parse the args even if I have a `-t` option, `timem -e PAPI_TOT_INS -- ls -t` only parses `-e PAPI_TOT_INS` and drops those args + the `--`, `timem -e PAPI_TOT_INS -- lldb -- sleep 2` still only parses `-e PAPI_TOT_INS`, etc.

File: cmake/fake_tribits.cmake (hunk 5 above)
Comment (jrmadsen) [reply cmake/fake_tribits.cmake:124]: No they aren't used in conjunction anywhere in the tests but I needed a way to pass more than one argument to one test. `ARGS` assumes that two arguments means two separate tests w/ the same executable w/ each one of the arguments.

File: core/src/impl/Kokkos_Core.cpp (hunk 6 above)
Comment (dalg24) [reply core/src/impl/Kokkos_Core.cpp:280]: Sorry I meant "call `std::exit` with a non-zero exit code".

Unconditionally exiting on `--help` can pose problem downstream though because we force user code to parse arguments before calling `Kokkos::initialize` so they can print their own help information.  Suppose the code is using another library that also exit on `--help` such as Google Benchmark and life becomes even more complicated.

Comment (DavidPoliakoff) [reply core/src/impl/Kokkos_Core.cpp:280]: Honestly I'd like to resolve this conversation in the weekly meeting. I agree with Jonathan's take on the subject, but it is a change from what other parts of Kokkos do, so we should discuss as a group

File: core/src/impl/Kokkos_Profiling.cpp (hunk 8 above)
[the remark this answers is not in the document]
Comment (DavidPoliakoff) [reply core/src/impl/Kokkos_Profiling.cpp:396]: Lets follow the path the rest of Core took, I guess. If both are set, check that they're equal

File: core/src/impl/Kokkos_Core.cpp (hunk 6 above)
Comment (jrmadsen) [reply core/src/impl/Kokkos_Core.cpp]: Resolved to call `std::exit(EXIT_SUCCESS)` for `--kokkos-tools-help`. `--kokkos-help` behavior is not currently being changed

File: core/src/impl/Kokkos_Profiling.cpp (hunk 8 above)
Comment (dalg24) [inline core/src/impl/Kokkos_Profiling.cpp:396]: This is now a bit outdated.
We should be doing
```C++
if (profileLibrary.empty())
  return;

char* envProfileLibrary = const_cast<char*>(profileLibrary.c_str());

File: core/src/impl/Kokkos_Profiling.hpp [hunk 10]
```diff
@@ -515,7 +520,12 @@ void beginDeepCopy(const SpaceHandle dst_space, const std::string dst_label,
                    const uint64_t size);
 void endDeepCopy();
 void finalize();
-void initialize();
+void initialize(const std::string& = {});
```
Comment (dalg24) [inline core/src/impl/Kokkos_Profiling.hpp:523]: Wasn't this forward declaration a duplicate?

File: core/src/impl/Kokkos_Profiling.hpp [hunk 11]
```diff
@@ -125,15 +125,20 @@ void syncDualView(const std::string& label, const void* const ptr,
 void modifyDualView(const std::string& label, const void* const ptr,
                     bool on_device);
 
-void initialize();
+void initialize(const std::string& = {});
 void finalize();
+bool printHelp(const std::string&);
+void parseArgs(const std::string&);
+void parseArgs(int, char**);
```
Comment (dalg24) [inline core/src/impl/Kokkos_Profiling.hpp:132]: Why do we need to expose this overload?

Comment (jrmadsen) [reply core/src/impl/Kokkos_Profiling.hpp:132]: - `void parseArgs(const std::string&)` for internal use
- `void parseArgs(int, char**)` for the user to call directly
  - I added this in the original version of the testing bc it excessively tedious to convert to a string when calling directly

File: core/src/impl/Kokkos_Profiling.hpp (hunk 10 above)
Comment (jrmadsen) [reply core/src/impl/Kokkos_Profiling.hpp:523]: No, there is a `Kokkos::Profiling::initialize(...)` and a `Kokkos::Tools::initialize(...)`

File: core/src/impl/Kokkos_Profiling.cpp (hunk 8 above)
Comment (dalg24) [reply core/src/impl/Kokkos_Profiling.cpp]: I expected you'd remove https://github.com/kokkos/kokkos/blob/a64a75f64d4113dfcf16a62bc12a9cb52b50fd19/core/src/impl/Kokkos_Profiling.cpp#L398-L402

**Review by crtrott — APPROVED** [review]

File: core/src/impl/Kokkos_Profiling.cpp (hunk 8 above)
Comment (dalg24) [reply core/src/impl/Kokkos_Profiling.cpp]: Ping

File: core/src/impl/Kokkos_Profiling.hpp (hunk 11 above)
Comment (DavidPoliakoff) [reply core/src/impl/Kokkos_Profiling.hpp:132]: I know I'm slow on this, but I just checked, and it doesn't look the tests call these. Do we still need both? Could we remove the int,char** guy?

**Review by DavidPoliakoff — APPROVED** [review]
LGTM
